src/reverse-proxy-handler.ts

```diff
@@ -71,6 +73,28 @@ function copyReverseOwnDescriptors(env: SecureEnvironment, shadowTarget: Reverse
     }
 }
 
+function callWithErrorBoundaryProtection(env: SecureEnvironment, fn: () => RawValue): RawValue {
+    let raw;
+    try {
+        raw = fn();
+    } catch (e) {
+        // This error occurred when the outer realm invokes a function from the sandbox.
+        if (e instanceof Error) {
+            throw e;
+        }
+        let rawError;
+        try {
+            rawError = new (env.getRawValue(e.constructor))(e.message);
+        } catch (ignored) {
+            // in case the constructor inference fails
+            rawError = new Error(e.message)
+        }
+        // by throwing a new raw error, we eliminate the stack information from the sandbox
+        throw rawError;
+    }
+    return raw;
+}
```

Comment:
This is a neat approach. In my work in this area I had a view places where internal code could throw so trapping was a bit easier so I could mask the error when caught and then re-throw. I leveraged pseudo-private mechanisms in Node to mark an error as being decorated and then also scrubbed it's stack. My error helpers [are here](https://github.com/standard-things/esm/tree/master/src/error). 

I leveraged V8 APIs (available in Chrome too) to help with this:
https://v8.dev/docs/stack-trace-api#stack-trace-collection-for-custom-exceptions

Reply:
Yeah, this definitely need more work, for now I'm kinda duct-taping this to try to get the right constructor, but we can iron this out later on.